src/fd_table.c

```diff
@@ -134,15 +137,20 @@ static uvwasi_errno_t uvwasi__get_type_and_rights(uv_file fd,
 
     *rights_base = UVWASI__RIGHTS_SOCKET_BASE;
     *rights_inheriting = UVWASI__RIGHTS_SOCKET_INHERITING;
+#endif
+#ifdef S_ISFIFO
```

Comment:
Same question about `S_IFIFO` and `S_IFBLK` on Windows.

Reply:
They don't exist, but I am not 100% sure about the reason. I believe Windows does not treat pipes and block devices as files either, but I might be wrong.